src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.h

```diff
@@ -17,8 +17,76 @@
 
 #pragma once
 
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
 #include <app/util/af-types.h>
 #include <app/util/basic-types.h>
+#include <app/util/config.h>
+#include <lib/core/DataModelTypes.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OccupancySensing {
+
+/**
+ * Interface to help manage the Hold Time Limits of the Occupancy Sensing Cluster.
+ */
+class HoldTimeLimitsManager : public AttributeAccessInterface
+{
+public:
+    static constexpr size_t kOccupancySensingServerMaxEndpointCount =
+        MATTER_DM_OCCUPANCY_SENSING_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+    static_assert(kOccupancySensingServerMaxEndpointCount <= kInvalidEndpointId, "Occupancy Sensing endpoint count error");
+
+    // HoldTimeLimits
+    class HoldTimeLimits
+    {
+    public:
+        Structs::HoldTimeLimitsStruct::Type * GetHoldTimeLimitsStruct(EndpointId endpoint);
+        CHIP_ERROR SetHoldTimeLimitsStruct(EndpointId endpoint, Structs::HoldTimeLimitsStruct::Type & holdTimeLimitsStruct);
+
+    private:
+        /// @brief Returns the index of the HoldTimeLimits associated to an endpoint
+        /// @param[in] endpoint target endpoint
+        /// @param[out] endpointIndex index of the corresponding HoldTimeLimits for an endpoint
+        /// @return CHIP_NO_ERROR or CHIP_ERROR_NOT_FOUND, CHIP_ERROR_INVALID_ARGUMENT if invalid endpoint
+        CHIP_ERROR FindHoldTimeLimitsIndex(EndpointId endpoint, size_t & endpointIndex);
+
+        Structs::HoldTimeLimitsStruct::Type mHoldTimeLimitsStructs[kOccupancySensingServerMaxEndpointCount];
+    };
+
+    static HoldTimeLimitsManager & Instance();
+
+    CHIP_ERROR Init();
+
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    // HoldTimeLimitsStruct Accessors
+    Structs::HoldTimeLimitsStruct::Type * GetHoldTimeLimitsStruct(EndpointId endpoint);
+    CHIP_ERROR SetHoldTimeLimitsStruct(EndpointId endpoint, Structs::HoldTimeLimitsStruct::Type & holdTimeLimitsStruct);
+
+private:
+    HoldTimeLimitsManager() : AttributeAccessInterface(Optional<EndpointId>(), Id) {}
+    ~HoldTimeLimitsManager() {}
+
+    bool mIsInitialized = false;
+
+    // HoldTimeLimits
+    HoldTimeLimits mHoldTimeLimits;
+
+    // Instance
+    static HoldTimeLimitsManager mInstance;
+};
+
+inline bool HasFeature(EndpointId ep, Feature feature)
+{
+    uint32_t map;
+    bool success = (Attributes::FeatureMap::Get(ep, &map) == Protocols::InteractionModel::Status::Success);
```

Comment:
Do not use global attribute getter for the features. Make the specific example have to initialize an instance with a `chip::BitFlags<>` of the technology features, from the initialization code, rather than using this round-about way of reading features which makes the ZAP file the source of truth in a way that is harder to understand than code, and which prevents one example from supporting multiple technologies.

Reply:
Changed in new code, please check